src/Client/PsrClickHouseClient.php

```diff
@@ -0,0 +1,176 @@
+use SimPod\ClickHouseClient\Output\Output;
+use SimPod\ClickHouseClient\Sql\ValueFormatter;
+use function array_key_first;
+use function array_keys;
+use function array_map;
+use function implode;
+use function is_int;
+use function Safe\sprintf;
+
+class PsrClickHouseClient implements ClickHouseClient
+{
+    /** @var ClientInterface */
+    private $client;
+
+    /** @var RequestFactory */
+    private $requestFactory;
+
+    /** @var LoggerInterface */
+    private $logger;
+
+    /** @var string */
+    private $endpoint;
+
+    /** @var array<string, float|int|string> */
+    private $defaultParameters;
+
+    /**
+     * @param array<string, float|int|string> $defaultParameters
+     */
+    public function __construct(
+        ClientInterface $client,
+        RequestFactory $requestFactory,
+        LoggerInterface $logger,
+        string $endpoint,
+        array $defaultParameters = []
+    ) {
+        $this->client            = $client;
+        $this->requestFactory    = $requestFactory;
+        $this->logger            = $logger;
+        $this->endpoint          = $endpoint;
+        $this->defaultParameters = $defaultParameters;
+    }
+
+    public function executeQuery(string $sql) : void
+    {
+        $response = $this->executeRequest($sql);
+
+        $contents = (string) $response->getBody();
+
+        return;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function select(string $sql, Format $outputFormat, array $requestParameters = []) : Output
+    {
+        $formatClause = $outputFormat::toSql();
+
+        $response = $this->executeRequest(
+            <<<CLICKHOUSE
+$sql
+$formatClause
+CLICKHOUSE,
+            $requestParameters
+        );
+
+        return $outputFormat::output((string) $response->getBody());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function insert(string $table, array $values, ?array $columns = null) : void
+    {
+        if ($values === []) {
+            throw CannotInsert::noValues();
+        }
+
+        if ($columns === null) {
+            $firstRow = $values[array_key_first($values)];
+            $columns  = array_keys($firstRow);
+            if (is_int($columns[0])) {
+                $columns = null;
+            }
+        }
+
+        $columnsSql = $columns === null ? '' : sprintf('(%s)', implode(',', $columns));
+
+        $valuesSql = implode(
+            ',',
+            array_map(
+                static function (array $map) : string {
+                    return sprintf(
+                        '(%s)',
+                        implode(',', ValueFormatter::formatMap($map))
+                    );
+                },
+                $values
+            )
+        );
+
+        $response = $this->executeRequest(
+            <<<CLICKHOUSE
+INSERT INTO $table
+$columnsSql
+VALUES $valuesSql
+CLICKHOUSE
+        );
+    }
+
+    public function insertWithFormat(string $table, Format $inputFormat, string $data) : void
```

Comment:
Kdy bys tohle použil? To je když máš třeba "raw data" v jsonu?

Reply:
jj https://clickhouse.yandex/docs/en/interfaces/formats/#inserting-data

Reply:
to je hrozná prasárna tohle ... ale ok

Reply:
docela stylovka za me

Reply:
json by byl ok, ale tohle "něco mezi", kde je separe json pro každej řádek, je prasárna